jupyterhub/user.py

```diff
@@ -196,11 +199,21 @@ def url(self):
 
     @gen.coroutine
     def spawn(self, options=None):
-        """Start the user's spawner"""
+        """Start the user's spawner
+        
+        Because there could be more then one server per user
+        each server has to have a unique reference (UUID4)
+        
+        base_url is built using user's base url and adding /servers/{name}
+        where name is the server uuid urlsafed
+        """
         db = self.db
+        server_uuid = uuid4()
+        server_url = urlsafe_b64encode(server_uuid.bytes)
```

Comment:
hex is url-safe. But it's also unique on a scale we really don't need to deal with (users aren't going to have 2^10 servers). See the above comment about using simple counters.